src/Functions/map.cpp

```diff
@@ -178,67 +181,88 @@ class FunctionMapFromArrays : public IFunction
                 getName(),
                 arguments.size());
 
-        /// The first argument should always be Array.
-        /// Because key type can not be nested type of Map, which is Tuple
-        DataTypePtr key_type;
-        if (const auto * keys_type = checkAndGetDataType<DataTypeArray>(arguments[0].get()))
-            key_type = keys_type->getNestedType();
-        else
-            throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "First argument for function {} must be an Array", getName());
+        auto get_nested_type = [this](const DataTypePtr & type) -> DataTypePtr
+        {
+            DataTypePtr nested;
+            if (const auto * array_type = checkAndGetDataType<DataTypeArray>(type.get()))
+                nested = array_type->getNestedType();
+            else if (const auto * map_type = checkAndGetDataType<DataTypeMap>(type.get()))
+                nested = std::make_shared<DataTypeTuple>(map_type->getKeyValueTypes());
+            else
+                throw Exception(
+                    ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                    "Argument types of function {} must be Array or Map, but {} is given",
+                    getName(),
+                    type->getName());
 
-        DataTypePtr value_type;
-        if (const auto * value_array_type = checkAndGetDataType<DataTypeArray>(arguments[1].get()))
-            value_type = value_array_type->getNestedType();
-        else if (const auto * value_map_type = checkAndGetDataType<DataTypeMap>(arguments[1].get()))
-            value_type = std::make_shared<DataTypeTuple>(value_map_type->getKeyValueTypes());
-        else
-            throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Second argument for function {} must be Array or Map", getName());
+            return nested;
+        };
 
+        auto key_type = get_nested_type(arguments[0]);
+        auto value_type = get_nested_type(arguments[1]);
+
+        /// Remove Nullable from key_type if needed for map key must not be Nullable
+        key_type = removeNullableOrLowCardinalityNullable(key_type);
```

Comment:
There is no such restrictions on value_type that it should not contain null values because Nullable is allowed as value type in map.  e.g.  for `mapFromArrays(Array(Nullable(K)), Array(Nullable(V)))`, the return type should be `Map(K, Nullable(V))` instead of `Map(K, V)`